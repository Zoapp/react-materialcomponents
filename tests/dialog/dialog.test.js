import React from "react";
import renderer from "react-test-renderer";
import dialogPolyfill from "dialog-polyfill";
import Dialog from "@libs/dialog/dialog";
import DialogFooter from "@libs/dialog/footer";

jest.mock("dialog-polyfill", () => jest.fn());

describe("dialog/Dialog", () => {
  it("renders correctly", () => {
    dialogPolyfill.registerDialog = jest.fn();

    // We are interested in seeing whether this method is called.
    const showModalSpy = jest.fn();

    // This represents the fake `dialogRef`
    const createNodeMock = () => ({
      addEventListener: jest.fn(),
      showModal: showModalSpy,
    });

    const component = renderer.create(
      <Dialog
        header="Are you happy?"
        actions={[{ name: "Cancel" }, { name: "Continue" }]}
      >
        <div>Hello, here is some content</div>
      </Dialog>,
      { createNodeMock }
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(showModalSpy).toHaveBeenCalled();
    expect(dialogPolyfill.registerDialog).not.toHaveBeenCalled();
  });

  it("uses the dialog-polyfill when needed", () => {
    dialogPolyfill.registerDialog = jest.fn();

    // This represents the fake `dialogRef`
    const createNodeMock = () => ({
      addEventListener: jest.fn(),
      // we simulate a `<dialog>` without the `showModal` function, which
      // implies the activation of the polyfill.
      // Because we cannot update the fake `dialogRef` after, we do not open
      // the Dialog by default (hence the prop `open={false}` below.
      showModal: undefined,
    });

    const component = renderer.create(
      <Dialog
        header="Are you happy?"
        actions={[{ name: "Cancel" }, { name: "Continue" }]}
        open={false}
      >
        <div>Hello, here is some content</div>
      </Dialog>,
      { createNodeMock }
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(dialogPolyfill.registerDialog).toHaveBeenCalled();
  });

  it("renders a dialog with no actions specified", () => {
    // This represents the fake `dialogRef`
    const createNodeMock = () => ({
      addEventListener: jest.fn(),
      showModal: jest.fn(),
    });

    const component = renderer.create(
      <Dialog
        header="Are you happy?"
      >
        <div>Hello, here is some content</div>
      </Dialog>,
      { createNodeMock }
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
