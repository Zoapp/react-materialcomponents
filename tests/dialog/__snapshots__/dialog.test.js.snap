// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog/Dialog can render a TextField if \`field\` prop is supplied 1`] = `
<dialog
  className="rmdc-dialog mdc-dialog mdc-dialog--open"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="presentation"
>
  <div
    className="mdc-dialog__surface"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="presentation"
    style={Object {}}
  >
    <header
      className="mdc-dialog__header"
    />
    <section
      className="mdc-dialog__body"
    >
      <div
        className="rmdc-text-field-wrapper"
      >
        <div
          className="mdc-text-field mdc-text-field--upgraded"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <input
            aria-controls="unique-component-id-helper-text"
            className="mdc-text-field__input"
            defaultValue=""
            id="unique-component-id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
          />
          <label
            className="mdc-floating-label"
            focused="false"
            htmlFor="unique-component-id"
            style={Object {}}
          >
            field-1
          </label>
          <div
            className="mdc-text-field__bottom-line"
          />
        </div>
        <p
          aria-hidden="true"
          className="mdc-text-field-helper-text mdc-text-field-helper-text--validation-msg"
        >
          error message
        </p>
      </div>
    </section>
    <footer
      className="mdc-dialog__footer"
    >
      <button
        className="mdc-dialog__footer__button mdc-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        Ok
      </button>
    </footer>
  </div>
</dialog>
`;

exports[`dialog/Dialog can render a TextField if \`field\` prop is supplied 2`] = `
<dialog
  className="rmdc-dialog mdc-dialog mdc-dialog--open"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="presentation"
>
  <div
    className="mdc-dialog__surface"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="presentation"
    style={Object {}}
  >
    <header
      className="mdc-dialog__header"
    />
    <section
      className="mdc-dialog__body"
    >
      <div
        className="rmdc-text-field-wrapper"
      >
        <div
          className="mdc-text-field mdc-text-field--upgraded mdc-text-field--invalid"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <input
            aria-controls="unique-component-id-helper-text"
            className="mdc-text-field__input"
            defaultValue=""
            id="unique-component-id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
          />
          <label
            className="mdc-floating-label"
            focused="false"
            htmlFor="unique-component-id"
            style={Object {}}
          >
            field-1
          </label>
          <div
            className="mdc-text-field__bottom-line"
          />
        </div>
        <p
          aria-hidden="true"
          className="mdc-text-field-helper-text mdc-text-field-helper-text--validation-msg"
        >
          error message
        </p>
      </div>
    </section>
    <footer
      className="mdc-dialog__footer"
    >
      <button
        className="mdc-dialog__footer__button mdc-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        Ok
      </button>
    </footer>
  </div>
</dialog>
`;

exports[`dialog/Dialog renders a dialog with no actions specified 1`] = `
<dialog
  className="rmdc-dialog mdc-dialog mdc-dialog--open"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="presentation"
>
  <div
    className="mdc-dialog__surface"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="presentation"
    style={Object {}}
  >
    <header
      className="mdc-dialog__header"
    >
      Are you happy?
    </header>
    <section
      className="mdc-dialog__body"
    >
      <div>
        Hello, here is some content
      </div>
    </section>
    <footer
      className="mdc-dialog__footer"
    >
      <button
        className="mdc-dialog__footer__button mdc-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        Ok
      </button>
    </footer>
  </div>
</dialog>
`;

exports[`dialog/Dialog renders correctly 1`] = `
<dialog
  className="rmdc-dialog mdc-dialog mdc-dialog--open"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="presentation"
>
  <div
    className="mdc-dialog__surface"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="presentation"
    style={Object {}}
  >
    <header
      className="mdc-dialog__header"
    >
      Are you happy?
    </header>
    <section
      className="mdc-dialog__body"
    >
      <div>
        Hello, here is some content
      </div>
    </section>
    <footer
      className="mdc-dialog__footer"
    >
      <button
        className="mdc-dialog__footer__button mdc-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        Continue
      </button>
      <button
        className="mdc-dialog__footer__button mdc-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        Cancel
      </button>
    </footer>
  </div>
</dialog>
`;

exports[`dialog/Dialog uses the dialog-polyfill when needed 1`] = `
<dialog
  className="rmdc-dialog mdc-dialog"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="presentation"
>
  <div
    className="mdc-dialog__surface"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="presentation"
    style={Object {}}
  >
    <header
      className="mdc-dialog__header"
    >
      Are you happy?
    </header>
    <section
      className="mdc-dialog__body"
    >
      <div>
        Hello, here is some content
      </div>
    </section>
    <footer
      className="mdc-dialog__footer"
    >
      <button
        className="mdc-dialog__footer__button mdc-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        Continue
      </button>
      <button
        className="mdc-dialog__footer__button mdc-button"
        disabled={false}
        id={null}
        onClick={[Function]}
        type="button"
      >
        
        Cancel
      </button>
    </footer>
  </div>
</dialog>
`;
